pkg/controller/machine/controller.go

```diff
@@ -323,3 +323,11 @@ func (r *ReconcileMachine) deleteNode(ctx context.Context, name string) error {
 	}
 	return r.Client.Delete(ctx, &node)
 }
+
+func delayAfterDrainFail(err error) (reconcile.Result, error) {
+	switch t := err.(type) {
+	case *controllerError.RequeueAfterError:
+		return reconcile.Result{Requeue: true, RequeueAfter: t.RequeueAfter}, nil
```

Comment:
Can't we just do it directly in `drainNode` method? There is only once place where `controllerError.RequeueAfterError` error types is returned.

Reply:
@ingvagabund yes, we could do that.  Unfortunately, no one wrote any unit tests for that function, and it's outside the scope of this change to do so.  Thus, this new code is tested and we can always consolidate later.

Reply:
`drainNode` is actually not tested. You are validating `delayAfterDrainFail` function and dogfooding it with expected data. We would need to mock part of `drainNode` method. Though, it requires to rewrite the machine controller a bit :-|. Which is even bigger pita.

Reply:
I would be in favor of dropping the test and just fixing https://github.com/openshift/cluster-api/pull/25/files#diff-b96ce5bf8686fb02d855188000ae1cb4R259 line. We need to verify `drainNode` is really re-queuing when the node draining itself fails, not the wrapper around it.

Reply:
@ingvagabund I'm not in favor of making a larger change than necessary here because we're in feature freeze.  I'm attempting to only modify the behavior for one specific condition, which I think this patch set does achieve.

I'd rather have an extra tested function than a 1 line change without tests.  We're short on tests as-is.  The test case is a nice implementation of the expected interface of drainNode; delay on node-drain, don't on all other failure types.

Reply:
Very well then.